File: PART3/ch02/ch02_01/java-kotlin/src/main/java/org/example/ex05/Exam05.java
package org.example.ex05;

import java.util.List;
import java.util.function.Consumer;
import java.util.function.Predicate;
import java.util.stream.Collectors;

public class Exam05 {

    //private Predicate<String> stringPredicate = new Predicate<String>() {
    //        @Override
    //        public boolean test(String s) {
    //            return s.equals("?");
    //        }
    //    };

    public Exam05(){
        // 고차함수

        var strList = List.of(
            "1",
            "2",
            "홍길동",
            "함수",
            "메소드"
        );

        var result = strList.stream()
            .filter(s -> s.equals(""))
            .collect(Collectors.toList());

    }

    public static void main(String[] args){
        new Exam05();
    }
}

File: PART3/ch01/kotlin-example/kotlin-example/src/main/kotlin/ex05/Exam05.kt
package ex05

import java.util.function.Predicate

fun main(){

    var numberList = listOf<Int>(1,2,3,4,5)

    val pred = Predicate<Int> { t -> t % 2 == 0 }

    numberList.stream().filter(pred)


    val add = { x: Int, y:Int -> x+y }

    val _add = fun(x: Int, y: Int): Int {
        return x+y
    }

    println(_add(3,4))
    println(add(2,3))

    lambda(4,5,_add)
}

fun lambda(x: Int, y:Int, method: (Int, Int) -> Int){
    println(method(x,y))
}